#!/usr/bin/env python
PACKAGE = "rover_nav"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Types: int_t, double_t, str_t & bool_t
map_max = 7;
map_min = -1 * map_max;
gen.add("start_x", double_t, 0, "g factor", 0, map_min, map_max)
gen.add("start_y", double_t, 0, "h factor", 0, map_min, map_max)
gen.add("goal_x", double_t, 0, "g factor", 1, map_min, map_max)
gen.add("goal_y", double_t, 0, "h factor", 1, map_min, map_max)
gen.add("g_gain", double_t, 0, "g factor", 1, 0)
gen.add("h_gain", double_t, 0, "h factor", 50, 0)
gen.add("elevation_gain", double_t, 0, "elevation factor", 40, 0)
gen.add("slope_gain", double_t, 0, "slope factor", 1, )

gen.add("elevation_layer", str_t, 0, "Elevation layer to use", "elevation")
gen.add("slope_layer", str_t, 0, "Slope layer to use", "normal_vectors_z")

exit(gen.generate(PACKAGE, "rover_nav", "planner")) 